lms/views/lti/oidc.py

```diff
@@ -0,0 +1,52 @@
+import uuid
+from urllib.parse import urlencode
+
+from pyramid.httpexceptions import HTTPForbidden, HTTPFound
+from pyramid.view import view_config
+
+from lms.services import LTIRegistrationService
+from lms.validation import OIDCRequestSchema
+
+
+@view_config(
+    route_name="lti.oidc",
+    request_method=("POST", "GET"),
+    schema=OIDCRequestSchema,
+)
+def oidc_view(request):
+    """
+    View for the second step of the OIDC authentication flow.
+
+    http://www.imsglobal.org/spec/security/v1p0/#step-2-authentication-request
+
+    We have to check we have a corresponding registration based on the `iss` and `client_id` params
+    and redirect back to the LMS.
+    """
+    params = request.parsed_params
+
+    registration = request.find_service(LTIRegistrationService).get(
+        issuer=params["iss"], client_id=params["client_id"]
+    )
+    if not registration:
+        raise HTTPForbidden(
+            f"""Registration not found. iss:'{params["iss"]}', client_id:'{params["client_id"]}'"""
+        )
+
+    params = {
+        # Hardcoded values from:
+        # http://www.imsglobal.org/spec/security/v1p0/#step-2-authentication-request
+        "scope": "openid",
+        "response_type": "id_token",
+        "response_mode": "form_post",
+        "prompt": "none",
+        "login_hint": params["login_hint"],
+        "lti_message_hint": params["lti_message_hint"],
+        "client_id": params["client_id"],
+        "state": uuid.uuid4().hex,
+        "nonce": uuid.uuid4().hex,
```

Comment:
These two are basically TODOs.

For nonce, if we want/need to check uniqueness we'll need a new table or some new volatile storage infra.

We might need to use it for LTI1.1 too: https://github.com/hypothesis/lms/blob/4da6be5175660cf361cfada5e5528cd2dd4e3224/lms/services/launch_verifier.py#L115


`state` is optional but we can tackle it in the same spike. Implementation might be simpler.

Reply:
Yeah, looks like we ignore the nonce elsewhere for the moment.